crates/rethnet_test_utils/src/lib.rs

```diff
@@ -1,12 +1,9 @@
-use rethnet_eth::remote::methods::ResolveUnspecifiedBlockTag;
-
-pub fn help_test_method_invocation_serde<MethodInvocation>(call: MethodInvocation)
-where
-    MethodInvocation: PartialEq
-        + std::fmt::Debug
-        + serde::de::DeserializeOwned
-        + serde::Serialize
-        + ResolveUnspecifiedBlockTag,
+#[allow(clippy::type_complexity)]
+pub fn help_test_method_invocation_serde<MethodInvocation>(
```

Comment:
Given your usage pattern, instead of using an `Option`, I'd create two methods, where one reuses the other:

`help_test_method_invocation_serde_with_expected(input: MethodInvocation, expected: MethodInvocation)`

```
fn help_test_method_invocation_serde(call: MethodInvocation) {
  help_test_method_invocation_serde_with_expected(call, call.clone())
}
```

Reply:
i like that! thank you! done in https://github.com/NomicFoundation/hardhat/pull/4132/commits/4a8cbaef09315e15c58c61fc03c47646704679af

Reply:
I don't actually see a single use case of the expected version of the function (apart from the helper function). Am I overlooking something in the commit?

Reply:
I looked at it in the commit view and the changes showed up there. Apologies. I think I had a view with just one file previously